### PR TITLE
140127375-nbsp-filter

### PR DESCRIPTION
**Pull request by benvand**
Required for https://www.pivotaltracker.com/story/show/140127375

* Add new template filter to convert spaces to nbsp
* Make available to all templates
* Add test
* Version bump

### CODE REVIEW

**Comment by benvand** [comment]
screen shot 2017-02-28 at 11 22 25
screen shot 2017-02-28 at 11 20 59

**Comment by benvand** [comment]
screen shot 2017-02-28 at 11 20 59

File: dmutils/filters.py [hunk 1]
```diff
@@ -40,3 +40,8 @@ def format_links(text):
         return formatted_text
     else:
         return text
+
+
+def nbsp(text):
+    """Replace spaces with nbsp."""
+    return Markup('&nbsp;'.join(text.split(' ')))
```
Comment (allait) [inline dmutils/filters.py]: We need to escape the original text before marking it as safe. This means we can't use this filter with anything that could contain HTML, but those parts should be `Markup` objects at this point anyway.

File: tests/test_filters.py [hunk 2]
```diff
@@ -66,3 +66,10 @@ def test_multiple_urls():
 def test_no_links_no_change():
     text = 'There are no Greek Γ Δ Ε Ζ Η Θ Ι Κ Λ links.'
     assert format_links(text) == text
+
+
+def test_nbsp():
```
Comment (allait) [inline tests/test_filters.py]: Let's add a test for escaping the original text too

File: dmutils/__init__.py [hunk 3]
```diff
@@ -3,4 +3,4 @@
 
 import flask_featureflags  # noqa
 
-__version__ = '24.0.4'
+__version__ = '24.0.5'
```
Comment (galund) [inline dmutils/__init__.py]: being pedantic, I think this is supposed to be a 24.1.0 as it's a new feature?

Comment (benvand) [inline dmutils/__init__.py]: Done 👌

**Pull request merged**